tests/test_port_add_remove.py

```diff
@@ -0,0 +1,136 @@
+                              num_of_ports-1,
+                              polling_config = PollingConfig(polling_interval = 1, timeout = 5.00, strict = True))
+
+        # verify that the port was removed properly since all buffer configuration was removed also
+        assert len(num) == num_of_ports - 1
+        
+        config_db.create_entry("PORT", PORT, port_info)
+
+    @pytest.mark.parametrize("scenario", ["one_port", "all_ports"])
+    def test_add_remove_all_the_ports(self, dvs, testlog, scenario):
+        config_db = dvs.get_config_db()
+        asic_db = dvs.get_asic_db()
+     
+        # get the number of ports before removal
+        num_of_ports = len(asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT"))
+        
+        # remove buffer pg cfg for the port
+        if scenario == "all_ports":
+            ports = config_db.get_keys('PORT')
+        elif scenario == "one_port":
+            ports = [PORT]
+        else:
+            assert False
+
+        ports_info = {}
+        
+        for i in range(DELETE_CREATE_ITERATIONS):
+            # remove ports
+            for key in ports:
+                # read port info and save it
+                ports_info[key] = config_db.get_entry("PORT", key)
+                
+                # remove a port
+                self.dvs_port.remove_port(key)
+    
+            # verify remove port
+            num = asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT",
+                                          num_of_ports-len(ports))
+            assert len(num) == num_of_ports-len(ports)
+            
+            # add port
+            time.sleep(i%3)
+            for key in ports:
+                config_db.create_entry("PORT", key, ports_info[key])
+
+            # verify add port            
+            num = asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT",
+                                  num_of_ports)
+            assert len(num) == num_of_ports
+                
+            time.sleep((i%2)+1)           
+            
+        # run ping
+        dvs.setup_db()        
+        dvs.create_vlan("6")
+        dvs.create_vlan_member("6", PORT)
+        dvs.create_vlan_member("6", "Ethernet68")
+        dvs.set_interface_status("Vlan6", "up")
+        dvs.add_ip_address("Vlan6", "6.6.6.1/24")
+        dvs.set_interface_status("Ethernet68", "up")
+        dvs.set_interface_status(PORT, "up")
+        
+        dvs.servers[16].runcmd("ifconfig eth0 6.6.6.6/24 up")
+        dvs.servers[16].runcmd("ip route add default via 6.6.6.1")
+        dvs.servers[17].runcmd("ifconfig eth0 6.6.6.7/24 up")
+        dvs.servers[17].runcmd("ip route add default via 6.6.6.1")
+        
+        time.sleep(2)
+        
+        rc = dvs.servers[16].runcmd("ping -c 1 6.6.6.7")
+        assert rc == 0
+
+        rc = dvs.servers[17].runcmd("ping -c 1 6.6.6.6")
+        assert rc == 0
+
+        dvs.remove_vlan_member("6", "Ethernet68")
```

Comment:
Change the PORT status (like admin etc) to whatever before the test case?

Reply:
changing the ports bach back to admin down was added

Reply:
Actually, we don't know if the port was admin down or not, I would think we should save the initial admin status, and then change to whatever it was before the test case.

Reply:
I noticed that on default it is down
I agree that the best thing to do is to save the initial admin status and change it back to the initial admin status in the end